src/go/k8s/pkg/resources/statefulset_update.go

```diff
@@ -264,27 +352,61 @@ func (r *StatefulSetResource) podEviction(ctx context.Context, pod, artificialPo
 		return err
 	}
 
-	var ordinal int64
+	var ordinal int32
 	ordinal, err = utils.GetPodOrdinal(pod.Name, r.pandaCluster.Name)
 	if err != nil {
 		return fmt.Errorf("cluster %s: cannot convert pod name (%s) to ordinal: %w", r.pandaCluster.Name, pod.Name, err)
 	}
 
-	if patchResult.IsEmpty() {
-		if err = r.checkMaintenanceMode(ctx, int32(ordinal)); err != nil {
+	managedDecommission, err := r.IsManagedDecommission()
+	if err != nil {
+		log.Error(err, "not performing a managed decommission")
+	}
+
+	isRedpandaConfigUpdate := utils.IsStatusPodConditionTrue(pod.Status.Conditions, ClusterUpdatePodCondition)
+
+	// if the pod definition changes, patchResult is not empty
+	// if this isn't a managedDecommission
+	// if the pod isn't needing to be updated for redpanda
+	// take the pod out of maintenance mode
+	if patchResult.IsEmpty() && !managedDecommission && !isRedpandaConfigUpdate {
+		if err = r.checkMaintenanceMode(ctx, ordinal); err != nil {
 			return &RequeueAfterError{
 				RequeueAfter: RequeueDuration,
 				Msg:          fmt.Sprintf("checking maintenance node (%s): %v", pod.Name, err),
 			}
 		}
-		return nil
+		return &RequeueAfterError{RequeueAfter: RequeueDuration, Msg: "wait for pod restart"}
```

Comment:
At this point you don't want to requeue, but rather move to next pod. If you return as in previous version `nil` the loop in `rollingUpdate` function will continue.

Reply:
I don't want to move to the next pod during this reconcile. Doing long-running loops is an antipattern for controllers.

Reply:
I just realised that this part of the code will be never called. 

Events are as follow:
1. put condition on all pods
2. delete pod aka remove condition from the pod
3. `rollingUpdate` will never create list of the already delete pod from step 2.